### DISCUSSION TITLE
Browser Back button not working on hash routes.

### DISCUSSION THREAD

**Discussion by Leomonay** [sveltejs/kit#4889] [Q&A]
I have been working in this repo a Svelte project that is deployed as a static website in ipfs:
https://github.com/anagolay/anagolay.network/tree/7-page-not-found-but-the-url-is-correct

(This is the branch where I want to fix this)

One of my navBar has the following links:

```
...
  <a href="/how-it-works><img src="main logo"/></a>
  <a href="/how-it-works>How it works</a>
  <a href="/index.html#useCases>Use Cases</a>
  ...
```

Steps to reproduce the bug:
```
1. Click on navbar "Use Cases" link.
2. Click on navbar "How it works" link.
3. Click browser back button.
```

This should lead to `/index.html#useCases`. Instead, it leads to the `error page`.

But the URL is correct. If the page is refreshed, It loads `index.html`

**Comment by Spenhouet** [comment]
I had a similar problem but in my case the cause was the manipulation of the window history:

```js
window.history.replaceState({}, '', '#useCases');
```

I didn't notice that with the first parameter set to `{}`, I'm not properly tracking the window history state.
This fixed it for me:

```js
window.history.replaceState(history.state, '', '#useCases');
```
